testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToHudi.java

```diff
@@ -0,0 +1,448 @@
+                .collect(toImmutableList()));
+
+        // just for symmetry
+        assertThat(second).containsOnly(first.rows().stream()
+                .map(QueryAssert.Row::new)
+                .collect(toImmutableList()));
+    }
+
+    private void createHudiCowTable(String tableName, Boolean partitioned)
+    {
+        createHudiTable(tableName, HUDI_TABLE_TYPE_COPY_ON_WRITE, partitioned);
+    }
+
+    private void createHudiMorTable(String tableName, Boolean partitioned)
+    {
+        createHudiTable(tableName, HUDI_TABLE_TYPE_MERGE_ON_READ, partitioned);
+    }
+
+    private void createHudiTable(String tableName, String tableType, Boolean partitioned)
+    {
+        if (partitioned) {
+            createHudiPartitionedTable(tableName, bucketName, tableType);
+            return;
+        }
+
+        createHudiNonPartitionedTable(tableName, bucketName, tableType);
+    }
+
+    private static void createHudiNonPartitionedTable(String tableName, String bucketName, String tableType)
+    {
+        onHudi().executeQuery(format(
+                """
+                        CREATE TABLE %s (
+                          id bigint,
+                          name string,
+                          price int,
+                          ts bigint)
+                        USING hudi
+                        TBLPROPERTIES (
+                          type = '%s',
+                          primaryKey = 'id',
+                          preCombineField = 'ts')
+                        LOCATION 's3://%s/%s'""",
+                tableName,
+                tableType,
+                bucketName,
+                tableName));
+
+        onHudi().executeQuery("INSERT INTO " + tableName + " VALUES (1, 'a1', 20, 1000), (2, 'a2', 40, 2000)");
+    }
+
+    private static void createHudiPartitionedTable(String tableName, String bucketName, String tableType)
+    {
+        onHudi().executeQuery(format(
+                """
+                        CREATE TABLE %s (
+                          id bigint,
+                          name string,
+                          ts bigint,
+                          dt string,
+                          hh string)
+                        USING hudi
+                        TBLPROPERTIES (
+                          type = '%s',
+                          primaryKey = 'id',
+                          preCombineField = 'ts')
+                        PARTITIONED BY (dt, hh)
+                        LOCATION 's3://%s/%s'""",
+                tableName,
+                tableType,
+                bucketName,
+                tableName));
+
+        onHudi().executeQuery("INSERT INTO " + tableName + " PARTITION (dt, hh) SELECT 1 AS id, 'a1' AS name, 1000 AS ts, '2021-12-09' AS dt, '10' AS hh");
+        onHudi().executeQuery("INSERT INTO " + tableName + " PARTITION (dt = '2021-12-09', hh='11') SELECT 2, 'a2', 1000");
+    }
+
+    private void createHudiSchema(String schemaName)
```

Comment:
I would inline this method.

Reply:
the same question: https://github.com/trinodb/trino/pull/15551#discussion_r1067928772

Reply:
The usage of this method is only one place. There's no benefit to extract a method.

Reply:
yes, once only. but in this case the extraction is to way make point attention that schema creates by specific not regular manner onHudi(), it's just an attempt to aggregate/extract onHudi() specifics in separate calls to prevent mix with onTrino() context.

Reply:
yes, once only. but in this case the extraction is to way make point attention that schema creates by specific not regular manner onHudi(), it's just an attempt to aggregate/extract onHudi() specifics in separate calls to prevent mix with onTrino() context.

Reply:
> it's just an attempt to/extract onHudi() specifics in separate calls to prevent mix with onTrino() context.

There's no need to do such efforts. I would recommend reading existing product tests.

Reply:
to summarise, is inlining of that is the change request or just optional change? ;)

Reply:
inlined